### ISSUE TITLE
Tracking: Remove feat/filestore branch Filecoin dependency

### ISSUE THREAD

**Issue by rvagg** [ipld/go-car#254]
#24 is a long-standing PR that has a branch that is relied on for Filecoin. It has the Blockstore API _and_ null padding, both of which are used by Filecoin. But we've never agreed to integrate these changes to master here so they have lingered for a ~year and we need to resolve it.

It's not believed that the Blockstore API is needed at this stage. But the null padding is. So this is the proposed plan:

 - [x] Pull out 32d0d9933 and add it as an option for the CARv1 API (@mvdan) - #171
 - [x] Tag a new version when ^ is merged
 - [ ] ~Switch to this new tag for go-fil-markets (@rvagg) - https://github.com/filecoin-project/go-fil-markets/pull/639 (draft)~
 - [x] Switch to this new tag for go-fil-markets (@rvagg) - https://github.com/filecoin-project/go-fil-markets/pull/709
 - [x] Switch to this new tag for lotus and fix the calling code to remove the 3rd return value (@rvagg) - https://github.com/filecoin-project/lotus/pull/7468 (draft)
 - [x] Remove go-commp-utils dependency on go-car https://github.com/filecoin-project/go-commp-utils/pull/7

**Comment by rvagg** [comment]
I've added https://github.com/filecoin-project/go-commp-utils/pull/7 to the list. It's in the lotus dependency tree and pulls in v0.1.1 of go-car even though it doesn't use it.

With that change, and switching go-fil-markets and lotus to go-car@master (as per the PRs linked above), I'm growing increasingly confident that we need neither of the commits on the custom branch. So I don't believe #255 is necessary. Only the most basic CAR workflows go through go-car, all of the storage/sealing/retrieval paths go through go-car/v2 which has the zero=eof option in-built and enabled.

I've got greens for all of my PRs, just using master here for v0. So I think the next step is to cut a v0.3.2 tag on master so the `SelectiveCar` changes can get a tag, and then go ahead and get those 3 PRs merged.

What to do with #255 is likely a separate question, will take it up over there.

**Comment by rvagg** [comment]
go-fil-markets update:

* we confirmed in https://github.com/filecoin-project/go-fil-markets/pull/639 that all we needed to do was switch to the master version, the filestore and zero==eof functionality had been removed by code evolution and adoption of go-car/v2.
* https://github.com/filecoin-project/go-fil-markets/pull/653 bulldozed an update to master in the meantime because it wanted `ctx` plumbing
* https://github.com/filecoin-project/go-fil-markets/pull/709 just updates to the tagged 0.3.3 as a minor cleanup.

**Comment by rvagg** [comment]
Lotus deps for go-car and go-commp-utils got updated; and it's got the latest go-fil-markets too so as per above comment it's not using the filestore branch. So I've ticked that off.

I'm pretty sure that "remove feat/filestore branch Filecoin dependency" has been completed. Getting the tagged version on go-fil-markets would be nice, but is not essential to resolve this. I'll try this week to get agreement to merge that but I'm going to close this out in the next couple of days unless someone sees a reason to disagree.
